### PR TITLE
[Clean] Eliminate registrations moved upstream.

### PR DESCRIPTION
**Pull request by dgeb**
Remove registrations moved into ember core via https://github.com/emberjs/ember.js/pull/13459

Cleanup of patches moved upstream into ember core will be done on the `clean` branch so that `master` can continue to accept patches and work in a backward compatible way as much as possible. Once all the patches are moved upstream, we'll merge `clean` to `master` and bump the minor version.

### CODE REVIEW

File: addon/-private/engine-ext.js [hunk 1]
```diff
@@ -34,73 +25,11 @@ Engine.reopen({
...
-      registry.injection('route', '_bucketCache', P`-bucket-cache:main`);
-      registry.injection('controller', '_bucketCache', P`-bucket-cache:main`);
-
-      registry.injection('route', 'router', 'router:main');
-
-      registry.register('component:-text-field', TextField);
-      registry.register('component:-text-area', TextArea);
-      registry.register('component:-checkbox', Checkbox);
-      // registry.register('view:-legacy-each', LegacyEachView);
       registry.register('component:link-to', LinkComponent);
```
Comment (rwjblue) [inline addon/-private/engine-ext.js]: Can you rename this one?  I keep getting confused if it is the normal `Ember.LinkComponent` or our custom scoped one.  Perhaps you can name the imported variable `EngineScopedLinkComponent`?

**Comment by rwjblue** [comment]
Left one nit-pick/tweak (non-blocking), but looks good to me!

File: addon/-private/engine-ext.js (hunk 1 above)
Comment (dgeb) [inline addon/-private/engine-ext.js]: Ah, good call. I get tripped up by that one too.

**Comment by dgeb** [comment]
Thanks @rwjblue!

**Pull request merged**